### ISSUE TITLE
Constructor with optional argument is not fully covered

### ISSUE THREAD

**Issue by thorny-thorny** [Kotlin/kotlinx-kover#139]
Hello! It seems there is an issue with constructors coverage. Constructor of class
```
class Greeter {
  constructor(
    name: String,
  ) {
    println(name)
  }
}
```
is 100% covered with simple test call `Greeter("Thorny")`. When I make a similar class with a default argument value so the argument is optional (`name: String = "world"`), I can't get it 100% covered no matter what I try:
```
GreeterWithOptional()
GreeterWithOptional("Thorny")
GreeterWithOptional(name = "Thorny")
```
Also I've tried to use `@JvmOverloads` and do the same tests in Java, but the report always says "Branch = 50% (1/2)". There can't be another branch, or do I miss something?

Kover plugin: 0.5.0, Kotlin/JVM plugin: 1.6.10, Gradle: 7.1

Code example: https://github.com/thorny-thorny/kover-tests/tree/optional-argument

Coverage report: [project-html.zip](https://github.com/Kotlin/kotlinx-kover/files/8054022/project-html.zip)

Thanks in advance!

**Comment by shanshin** [comment]
Thanks for the report!

**Comment by zuevmaxim** [comment]
@shanshin Could you please try IJCoverage version 1.0.660? The issue must be fixed then.

**Comment by thorny-thorny** [comment]
It was too late when I realized that it wasn't addressed to me :D So I've tried 1.0.660. It correctly shows 100% coverage for full tests. But for partial tests it shows different results: 100% (2/2) for `GreeterWithOptional()`-only test and 50% (1/2) for `GreeterWithOptional("Thorny")`-only test. I guess it should show 1/2 for both cases.

[branch](https://github.com/thorny-thorny/kover-tests/tree/optional-argument-1.0.660)

**Comment by zuevmaxim** [comment]
@thorny-thorny Hi! Thank you for your tests. Do you mean 1/2 method coverage? I have 0/0 branch coverage for all cases.

As for coverage of methods  with default arguments, we think that there should be 100% branch and method coverage, so that a user does not know that there is some generated code underneath.

**Comment by thorny-thorny** [comment]
@zuevmaxim Yes, I mean method coverage. Well I don't know much about testing stuff but I wouldn't mind to have 100% coverage only for test where both `GreeterWithOptional()` and `GreeterWithOptional("Thorny")` are present to force devs to test both cases and potentially catch some bugs

**Comment by zuevmaxim** [comment]
@thorny-thorny Could you please try version 1.0.661? Method coverage must be fixed.

**Comment by thorny-thorny** [comment]
@zuevmaxim 
Only `GreeterWithOptional()`: 1/1 methods, 3/3 lines
Only `GreeterWithOptional("Thorny")`: 1/1 methods, 2/3 lines
Both: 1/1 methods, 3/3 lines

**Comment by zuevmaxim** [comment]
@thorny-thorny Thank you! This is a correct behavior in my opinion.

**Comment by shanshin** [comment]
Fixed in `0.5.1`
